pkg/data/resource_metrics.go

```diff
@@ -0,0 +1,356 @@
+package data
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+)
+
+type MetricPropType string
+const (
+	USED    MetricPropType = "Used"
+	CAP     MetricPropType = "Capacity"
+	PEAK    MetricPropType = "Peak"
+	AVERAGE MetricPropType = "Average"
+)
+
+type ResourceType string
+const (
+	CPU      ResourceType = "CPU"
+	MEM      ResourceType = "MEM"
+	DISK     ResourceType = "DISK"
+	MEM_PROV ResourceType = "MEM_PROV"
+	CPU_PROV ResourceType = "CPU_PROV"
+)
+
+
+type EntityType string
+const (
+	NODE EntityType = "Node"
+	POD  EntityType = "Pod"
+	APP  EntityType = "App"
+)
+type ResourceMetric interface {
+	GetMetric() MetricPropType
+	GetResourceType() ResourceType
+	SetValue(data *float64)
+	GetValue() *float64
+}
+
+// ====================== Entity Resource Metric ===========================
+
+type EntityResourceMetrics struct {
+	entityId                string
+	ResourceMetricMap       map[string]ResourceMetric
+	boughtResourceMetricMap map[string]ResourceMetric
+}
+
+func NewEntityResourceMetrics(entityId string) *EntityResourceMetrics {
+	return &EntityResourceMetrics{
+		entityId: entityId,
+		ResourceMetricMap: make(map[string]ResourceMetric),
+	}
+}
+func (store *EntityResourceMetrics) SetResourceMetric(resource ResourceMetric) {
+	mkey := string(resource.GetResourceType()) + "::" + string(resource.GetMetric())
+	store.ResourceMetricMap[mkey] = resource
+}
+func (store *EntityResourceMetrics) GetResourceMetric(resourceType ResourceType, metricType MetricPropType) ResourceMetric {
+	if store == nil {
+		glog.V(4).Infof("Resource metric store not initialized")
+		return nil
+	}
+	mkey := string(resourceType) + "::" + string(metricType)
+	metric, ok := store.ResourceMetricMap[mkey]
+	if !ok {
+		glog.V(4).Infof("Cannot find metrics for %s\n", mkey)
+		return nil
+	}
+	return metric
+}
+
+func (store *EntityResourceMetrics) GetSoldResourceMetricSet() map[string]ResourceMetric {
+	return store.ResourceMetricMap
+}
+
+func (store *EntityResourceMetrics) printMetrics() {
+	fmt.Printf("======= Entity %s\n", store.entityId)
+	for mkey, resourceMetric := range store.ResourceMetricMap {
+		fmt.Printf("%s", mkey)
+		if resourceMetric.GetValue() != nil {
+			fmt.Printf(" %f\n", *resourceMetric.GetValue())
+		}
+	}
+}
+
+// ====================== Resource Metric ===========================
+type CPUUsedMetric struct {
+	used *float64
+}
+
+func NewCPUUsedMetric() *CPUUsedMetric {
+	return &CPUUsedMetric{}
+}
+func (cpu *CPUUsedMetric) GetResourceType() ResourceType {
+	return CPU
+}
+func (cpu *CPUUsedMetric) GetMetric() MetricPropType {
+	return USED
+}
+func (cpu *CPUUsedMetric) SetValue(data *float64) {
```

Comment:
why dont we set the value when we create the Metric. Or do we want to change the value of metric at some point?